interdiff_test.go

```diff
@@ -68,3 +83,52 @@ func TestInterDiffMode(t *testing.T) {
 		})
 	}
 }
+
+func TestApplyDiff(t *testing.T) {
+	for _, tt := range applyDiffFileTests {
+		t.Run(tt.resultFile, func(t *testing.T) {
+			source, err := ioutil.ReadFile(tt.sourceFile)
+
+			if err != nil {
+				t.Errorf("Error reading sourceFile %q", tt.sourceFile)
+			}
+
+			diffFile, err := os.Open(tt.diffFile)
+
+			if err != nil {
+				t.Errorf("Error opening diffFile %q", tt.diffFile)
+			}
+
+			var diffReader io.Reader = diffFile
+
+			diffFileDiff, err := diff.NewFileDiffReader(diffReader).Read()
```

Comment:
```suggestion
			d, err := diff.NewFileDiffReader(diffFile).Read()
```

You should be able to use `diffFile` as an `io.Reader` directly.

Also golang folks like short variable names, so maybe try calling this one just `d`.